### ISSUE TITLE
[v3] Implement new Concept Exercise: basics

### ISSUE THREAD

**Issue by wolf99** [exercism/c#586]
This issue describes how to implement the `<SLUG>` concept exercise for the <LANG> track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the student basic concepts of the C++ language.

## Learning objectives

- Know what a variable is.
- Know how to define a variable.
- Know how to update a variable.
- Know how to use type inference for variables.
- Know how to define a function.
- Know how to return a value from a function.
- Know how to call a function.
- Know about the static modifier.
- Know how to define an integer.
- Know how to use mathematical operators on integers.
- Know how to define single- and multiline comments.

## Out of scope

- Naming rules for identifiers.
- Memory and performance characteristics
- Preprocessor defines

## Concepts

- `basics`: know what a variable is; know how to define a variable; know how to update a variable; know how to define a function; know how to return a value from a function; know how to call a function;  know about the static modifier; know how to defined an integer; know how to use mathematical operators on integers; know how to define single and multiline comments.

## Prerequisites

There are no prerequisites.

### Hints

- How to define an integer
- How to define a function without any arguments
- How to return an integer from a function
- How to add or subtract integers using arithmetic operators
- How to invoke a function

### After

- More info on the assignment operators
- Introduction of the static keyword
- More information on scoping
- More info on commenting syntax

## Implementing

To implement this exercise, please follow [these instructions][implementation-instructions]

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

[implementation-instructions]: https://github.com/exercism/v3/blob/master/languages/c/reference/implementing-a-concept-exercise.md
